assets/js/src/marketing_optin_block/block.tsx

```diff
@@ -32,15 +32,8 @@ const Block = (
   }
 
   return (
-    <CheckboxControl
-      checked={checked}
-      onChange={setChecked}
-    >
-      {/* eslint-disable-next-line react/no-danger */}
-      <span dangerouslySetInnerHTML={{
-        __html: text || defaultText,
-      }}
-      />
+    <CheckboxControl checked={checked} onChange={setChecked}>
+      <RawHTML>{ text || defaultText }</RawHTML>
```

Comment:
I'd like to understand this. The `RawHTML` component just prints the contents just like the `dangerouslySetInnerHTML`, right? Is there a concern about XSS? Is the text sanitised?

Reply:
It is saved within a post and escaped the same way as any other post content.